### PR TITLE
91130 Update test coverage - form 10-7959a

### PR DESCRIPTION
**Pull request by michaelclement**
## Are you removing, renaming or moving a folder in this PR?
- [X] No, I'm not changing any folders (skip to Summary and delete the rest of this section)
- [ ] Yes, I'm removing, renaming or moving a folder

Did you change site-wide styles, platform utilities or other infrastructure?
- [X] No
- [ ] Yes, and I used the [proxy-rewrite steps](https://github.com/department-of-veterans-affairs/vets-website/tree/main/src/applications/proxy-rewrite#that-sounds-normal-so-whats-the-proxy-all-about) to test the injected header scenario

## Summary

Update unit and e2e tests for form 10-7959a.

- Adds the first E2E test
- Updates some unit tests, increasing coverage

## Related issue(s)

- https://github.com/department-of-veterans-affairs/va.gov-team/issues/91130

## Testing done

- Unit
- E2E

## Screenshots

|         | Before | After |
| ------- | ------ | ----- |
| Unit tests  |Screenshot 2024-08-21 at 10 21 03|Screenshot 2024-08-21 at 10 16 32|
| Cypress |NA|Screenshot 2024-08-21 at 11 00 24|

## What areas of the site does it impact?

*(Describe what parts of the site are impacted **if** code touched other areas)*

### Quality Assurance & Testing

- [x] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [x] Linting warnings have been addressed
- [ ] Documentation has been updated ([link to documentation](#) \*if necessary)
- [x] Screenshot of the developed feature is added
- [ ] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Error Handling

- [x] Browser console contains no warnings or errors.
- [x] Events are being sent to the appropriate logging solution
- [ ] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

- [ ] Did you login to a local build and verify all authenticated routes work as expected with a test user

## Requested Feedback

(OPTIONAL) _What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

File: src/applications/ivc-champva/10-7959a/config/submitTransformer.js [hunk 1]
```diff
@@ -49,7 +49,7 @@ export default function transformForSubmit(formConfig, form) {
   if (copyOfData.certifierAddress) {
     // Combine streets for 3rd party certifier
     copyOfData.certifierAddress.streetCombined = concatStreets(
-      copyOfData.certifierAddress.streetCombined,
+      copyOfData.certifierAddress,
```
Comment (michaelclement) [inline src/applications/ivc-champva/10-7959a/config/submitTransformer.js:52]: Fixing - this was a misuse of the `concatStreets` fn

File: src/applications/ivc-champva/10-7959a/config/submitTransformer.js [hunk 2]
```diff
@@ -61,6 +61,7 @@ export default function transformForSubmit(formConfig, form) {
     copyOfData.medicalUpload,
     copyOfData.primaryEob,
     copyOfData.secondaryEob,
+    copyOfData.pharmacyUpload,
```
Comment (michaelclement) [inline src/applications/ivc-champva/10-7959a/config/submitTransformer.js:64]: This file was missing previously, so it wasn't getting attached to the generated PDF. This fixes it.

File: src/applications/ivc-champva/10-7959a/containers/IntroductionPage.jsx [hunk 3]
```diff
@@ -9,7 +9,7 @@ import { VaLink } from '@department-of-veterans-affairs/component-library/dist/r
 export default function IntroductionPage(props) {
   const { route } = props;
   const { formConfig, pageList } = route;
-  const { appType } = formConfig?.customText || 'application';
+  const { appType } = formConfig?.customText;
```
Comment (michaelclement) [inline src/applications/ivc-champva/10-7959a/containers/IntroductionPage.jsx:12]: This was a red herring as the custom text argument was always defined, and didn't actually do what was originally intended, namely set a default value for `appType`. Removing for simplicity.

File: src/applications/ivc-champva/10-7959a/tests/10-7959a.cypress.spec.js
Comment (michaelclement) [inline src/applications/ivc-champva/10-7959a/tests/10-7959a.cypress.spec.js:1]: New cypress test to do a full run through of the form.

File: src/applications/ivc-champva/10-7959a/tests/unit/containers/ConfirmationPage.unit.spec.js [hunk 4]
```diff
@@ -24,7 +24,7 @@ const storeBase = {
   },
 };
 
-const storeBaseInvalid = {
+const storeBaseTruncated = {
```
Comment (michaelclement) [inline src/applications/ivc-champva/10-7959a/tests/unit/containers/ConfirmationPage.unit.spec.js:27]: Confusingly named, as it was valid data. Renamed to indicate that it's a shorter version of the test data used above.

**Review by cloudmagic80 — APPROVED** [review]

**Review by balexandr — APPROVED** [review]

**Review by Andrew565 — APPROVED** [review]

**Pull request merged**